### DISCUSSION TITLE
About the output build warning: JS library symbol '$allocateUTF8' is deprecated. Please open a bug if you have a continuing need for this symbol [-Wdeprecated]

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#21909] [Q&A]
emcc 3.1.55

My App is use in `allocateUTF8 `for converting JS String to c++:

c++ code:
```
void myFunc(char* strUtf8)
{
//do somthing with strUtf8
 free(strUtf8);
}
```

JS code:

`_myFunc(allocateUTF8("MyString"));`

What is alternative to `allocateUTF8 `? what are the changes that should be done for the above example ?

**Comment by sbc100** [comment]
The replacement for `allocateUTF8` is `stringToNewUTF8`.

**Comment by aharondavid** [comment]
Thanks!
